src/overlays/actors/ovl_En_Fall/z_en_fall.c

```diff
@@ -36,75 +64,820 @@ const ActorInit En_Fall_InitVars = {
+                break;
+            case EN_FALL_TYPE_MOONS_TEAR:
+                this->actor.update = EnFall_Update;
+                this->actor.draw = NULL;
+                this->actionFunc = EnFall_MoonsTear_Fall;
+                Actor_SetScale(&this->actor, 0.02f);
+                if ((globalCtx->actorCtx.unk5 & 2) == 0) {
+                    Actor_MarkForDeath(&this->actor);
+                }
+                moon = EnFall_MoonsTear_GetTerminaFieldMoon(globalCtx);
+                this->actor.child = moon;
+                if (moon == NULL) {
+                    Actor_MarkForDeath(&this->actor);
+                }
+                break;
+            case EN_FALL_TYPE_STOPPED_MOON_OPEN_MOUTH:
+                this->actor.draw = NULL;
+                this->actionFunc = EnFall_StoppedOpenMouthMoon_PerformCutsceneActions;
+                Actor_SetScale(&this->actor, this->scale * 3.0f);
+                break;
+            case EN_FALL_TYPE_CRASH_FIRE_RING:
+                this->actor.update = EnFall_FireRing_Update;
+                this->actor.draw = EnFall_FireRing_Draw;
+                Actor_SetScale(&this->actor, 0.2f);
+                break;
+            default:
+                this->actor.draw = EnFall_Moon_Draw;
+                this->dayStartTime = CLOCK_TIME(6, 0);
+                this->currentDay = CURRENT_DAY;
+                EnFall_Moon_AdjustScaleAndPosition(this, globalCtx);
+                break;
+        }
+    }
+}
+
+void EnFall_CrashingMoon_HandleGiantsCutscene(EnFall* this, GlobalContext* globalCtx) {
+    static s32 sGiantsCutsceneState = 0;
+
+    if (globalCtx->sceneNum == SCENE_00KEIKOKU && gSaveContext.sceneSetupIndex == 1 && globalCtx->csCtx.unk_12 == 0) {
+        switch (sGiantsCutsceneState) {
+            case 0:
+                if (globalCtx->csCtx.state != 0) {
+                    sGiantsCutsceneState += 2;
+                }
+                break;
+            case 2:
+                if (CHECK_QUEST_ITEM(QUEST_REMAINS_ODOWLA) && CHECK_QUEST_ITEM(QUEST_REMAINS_GOHT) &&
+                    CHECK_QUEST_ITEM(QUEST_REMAINS_GYORG) && CHECK_QUEST_ITEM(QUEST_REMAINS_TWINMOLD)) {
+                    if (gSaveContext.weekEventReg[0x5D] & 4) {
+                        if (ActorCutscene_GetCanPlayNext(0xC)) {
+                            ActorCutscene_Start(0xC, &this->actor);
+                            sGiantsCutsceneState++;
+                        } else {
+                            ActorCutscene_SetIntentToPlay(0xC);
+                        }
+                    } else if (ActorCutscene_GetCanPlayNext(0xB)) {
+                        ActorCutscene_Start(0xB, &this->actor);
+                        gSaveContext.weekEventReg[0x5D] |= 4;
+                        sGiantsCutsceneState++;
+                    } else {
+                        ActorCutscene_SetIntentToPlay(0xB);
+                    }
+                } else if (globalCtx->csCtx.frames >= 0x641) {
```

Comment:
```suggestion
                } else if (globalCtx->csCtx.frames > 1600) {
```
Is frames suppose to be in decimal or hex? I can never remember. Dec makes more sense to be logically, but go with however the rest of the code-base does it. Frames are used a few places around too

Reply:
Unfortunately, there's a degree of inconsistency. z_bg_ctower_rot and z_bg_iknv_obj use hex, but most other things use decimal. I'll switch to decimal here to be more consistent.